clang/lib/APINotes/APINotesManager.cpp

```diff
@@ -233,6 +234,12 @@ APINotesManager::getCurrentModuleAPINotes(Module *M, bool LookInModule,
 
         APINotes.push_back(*File);
       }
+      // If module FooCore is re-exported through module Foo, try Foo.apinotes.
+      if (!ExportedModuleName.empty())
+        if (auto File = findAPINotesFile(Dir, ExportedModuleName,
+                                         /*WantPublic=*/true)) {
```

Comment:
Why are we forcing public? Private modules can `export_as` too.

Reply:
Ah, wasn't aware of that, fixed